### PR TITLE
Adding support for multiple azure environments

### PR DESCRIPTION
**Pull request by tariq1890**
Currently, the azure discovery assumes the environment as AzurePublicCloud. Other supported Azure Environments won't with the current code.

This PR changes that.  Minor code improvements have been made as well.

### CODE REVIEW

File: discovery/azure/azure.go [hunk 1]
```diff
@@ -157,29 +159,45 @@ type azureClient struct {
 	vmssvm compute.VirtualMachineScaleSetVMsClient
 }
 
+// getAzureEnvironment returns an instance of azure.Environment based on envName.
+// If an invalid/unrecognized value of envName is passed, azurePublicCloud is returned by default.
```
Comment (brian-brazil) [inline discovery/azure/azure.go]: This should be an error, not silently failing.

**Comment by tariq1890** [comment]
@brian-brazil Thanks for the review. I will fix it in the morning. Is there any other comment that you for the PR?

**Comment by tariq1890** [comment]
Done!

File: docs/configuration/configuration.md [hunk 2]
```diff
@@ -272,6 +272,8 @@ See below for the configuration options for Azure discovery:
 
 ```yaml
 # The information to access the Azure API.
+# The Azure environment.
+environment: <string>
```
Comment (grobie) [inline docs/configuration/configuration.md]: This seems to be `[ environment: <string> | default = "AzurePublicCloud" ]` :-)

Comment (tariq1890) [reply docs/configuration/configuration.md]: oh yes :(. Thanks for pointing it out.

File: discovery/azure/azure.go [hunk 3]
```diff
@@ -157,29 +159,48 @@ type azureClient struct {
 	vmssvm compute.VirtualMachineScaleSetVMsClient
 }
 
+// getAzureEnvironment returns an instance of azure.Environment based on envName.
+func getAzureEnvironment(envName string) (azure.Environment, error) {
+	env, err := azure.EnvironmentFromName(envName)
```
Comment (grobie) [inline discovery/azure/azure.go]: The next 4 lines are the same as `return azure.EnvironmentFromName(envName)`. In the end I don't think we need this function, just use `azure.EnvironmentFromName(cfg.Environment)` below in `createAzureClient()`.

Comment (tariq1890) [reply discovery/azure/azure.go]: Ah yes. Done

**Pull request merged**